sdk/rust/src/processor/mod.rs

```diff
@@ -110,7 +116,7 @@ impl<'a> TransactionProcessor<'a> {
                     };
 
                 // Absorb the TpRegisterResponse message
-                let _ = match future.get(){
+                let _ = match future.get_timeout(Duration::from_millis(40000)) {
```

Comment:
Is the duration arbitrary?

Reply:
Yes, you looked at the PR before I added a message asking for better ways to handle this.

Reply:
In the other SDK's we don't time out on registration.

Reply:
It updated it to timeout out at 10 seconds (instead of 40). The timeout is required because the control handler will not stop the blocking on future.get. Instead, if registration is not completely in 10 seconds, registration will be retried if a control c had not been received.